test_suite/conftest.py

```diff
@@ -32,15 +39,36 @@ def check_markers(host, request):
     ]
 
     host_distro = host.system_info.distribution
-    host_version = host.system_info.release
+    host_version = float(host.system_info.release)
     distro_version = f'{host_distro}{host_version}'
 
     # Skip the test if the distro is explicitly excluded
     if exclude_on_marker:
         exclude_on_marker_list = exclude_on_marker.args[0]
         if host_distro in exclude_on_marker_list \
                 or distro_version in exclude_on_marker_list:
-            pytest.skip(f"This test doesn't apply to {distro_version}")
+            pytest.skip(skip_massage.format(distro_version=distro_version))
+
+        elif len(exclude_on_marker_list) > 1 and host_distro in exclude_on_marker_list[0]:
+            exclude_on_marker_list_version = parse_distro_version(exclude_on_marker_list[0])
+            if exclude_on_marker_list[1] == '<':
+                if host_version < exclude_on_marker_list_version:
+                    pytest.skip(f"Skip test with version smaller than {exclude_on_marker_list[0]}")
+            elif exclude_on_marker_list[1] == '>':
+                if host_version > exclude_on_marker_list_version:
+                    pytest.skip(f"Skip test with version bigger than {exclude_on_marker_list[0]}")
+
+    if major_version_marker:
+        major_version_marker_list = major_version_marker.args[0]
+        for major_version in major_version_marker_list:
+            major_version_marker_list_version = parse_distro_version(major_version)
+            if host_version >= major_version_marker_list_version:
+                if int(host_version) == int(major_version_marker_list_version):
```

Comment:
This if can be inverted so that you don't need to use "else".
```suggestion
                if int(host_version) != int(major_version_marker_list_version):
                    pytest.skip(skip_message.format(distro_version=distro_version))
```

Reply:
This is a possible way, although I thought about something else as well:
            if host_version >= major_version_marker_list_version \
                    and int(host_version) == int(major_version_marker_list_version):
                return
            else:
                pytest.skip(skip_massage.format(distro_version=distro_version))

what do you think?

Reply:
Yeah I was thinking about the same too! However let's see what's the outcome of this comment:
https://github.com/osbuild/cloud-image-val/pull/186#discussion_r1097132205